### PR TITLE
refactor: use button component

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-ksadjust-tutorial-meta-button-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-docker) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202016460311396) 🎟️

## What

This PR refactors the tutorial interactive lab button to align with the [latest designs](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Dev-Portal?node-id=6297%3A71281), using the `Button` component. 

Screen Shot 2022-04-04 at 4 01 14 PM

### CODE REVIEW

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { IconTerminalScreen16 } from '@hashicorp/flight-icons/svg-react/terminal-screen-16'
+import Button from 'components/button'
 import { useInstruqtEmbed } from 'contexts/instruqt-lab'
-import s from './interactive-lab-button.module.css'
 
-// @TODO replace with Button component once implemented
+/**@TODO this button shows a drop-shadow elevation in the design,
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/interactive-lab-button/index.tsx:5]: [See this ticket](https://app.asana.com/0/1202001387788447/1202073130822761)

Comment (ashleemboyer) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:5]: Just checked [the DS Button component](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/?node-id=2%3A9) for this, elevation is definitely missing on the primary style buttons! Also noticed that the elevation should go away in the `focus-visible` state, so those are listed here too.

I believe the updates that need to be made are:

- Add the `box-shadow` property to the `.primary` class:
  ```css
  box-shadow: var(--token-elevation-low-box-shadow);
  ```
- Set `box-shadow: none` in `.primary:focus-visible` state
- Set `box-shadow: none` in `.secondary:focus-visible` state
- Set `box-shadow: none` in `.critical:focus-visible` state

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx [hunk 2]
```diff
@@ -14,9 +16,13 @@ export default function InteractiveLabButton() {
...
-      <span>{buttonText}</span>
-    </button>
+    <Button
+      color="primary"
+      text={buttonText}
+      onClick={() => ctx.setActive(!ctx.active)}
+      size="medium"
+      iconPosition="leading"
+      icon={<IconTerminalScreen16 />}
+    />
```
Comment (ashleemboyer) [inline src/components/tutorial-meta/components/interactive-lab-button/index.tsx:26]: These values for `color`, `iconPosition`, and `size` are the default values, would you like to omit them?

**Review by zchsh — APPROVED** [review]
✅ ⏯️ Button looks good to me in the preview.

+1 on Ashlee's comment on updating `button`, if we can fit that into this PR that seems good, but also open to doing it in a follow-up.

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx (hunk 1 above)
Comment (didoo) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:5]: FYI elevation style in HDS Button is not applied for `active` and `focused` states: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components-%5BWIP%5D?node-id=2340%3A21001

Comment (ashleemboyer) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:5]: Thank you @didoo! We've got the active states covered ([see here](https://github.com/hashicorp/dev-portal/blob/a13b86ac4998ed58b1533c4891f21629834b15c3/src/components/button/button.module.css)), the focus ones are what we haven't handled yet. Just to check, the `disabled` states don't have elevation applied either right?

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx (hunk 2 above)
Comment (didoo) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:26]: Yes, `disabled` also doesn't have elevation (essentially elevation is only for `default` + `hover` :)

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx (hunk 1 above)
Comment (didoo) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:5]: Yes, `disabled` also doesn't have elevation (essentially elevation is only for `default` + `hover` :)

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx (hunk 2 above)
Comment (kendallstrautman) [reply src/components/tutorial-meta/components/interactive-lab-button/index.tsx:26]: Oh rad! ya I'll remove those :)

**Comment by kendallstrautman** [comment]
Alright @zchsh & @ashleemboyer, made those primary button box shadow updates. Lmk if it looks okay!

**Review by zchsh — APPROVED** [review]
✅ 💅 New preview looks good to me!

`box-shadow` is a nice _subtle_ touch, I for one had to check in dev-tools that it's there 🤣 💯 looks great ☺️

**Comment by ashleemboyer** [comment]
> @ashleemboyer good call, I _think_ that [should happen here](https://github.com/hashicorp/dev-portal/blob/main/src/components/lab-embed/embed-element/index.tsx#L14-L16), but we should test it to make sure! We can make a follow-up ticket if it's not working as expected 👌

@kendallstrautman that's helpful to see that code, thank you! I'm going to file a bug report then since we expect it to work but it isn't. I will fill in the details to reproduce and will attach a screen recording here: https://app.asana.com/0/1201987349274776/1202074017129372/f

**Pull request merged**